File: hw/chavez/u3/hw25TemplateMethod/Python/view/beverage_test.py
import sys
import os
#sys.path.append(os.path.dirname(os.path.dirname(__file__)))
from model.coffee import Coffee
from model.tea import Tea
1
if __name__ == "__main__":
    option = 0
    tea = Tea()
    coffee = Coffee()

    while option != 3:
        print("--- Beverage Menu ---")
        print("1. Make Tea")
        print("2. Make Coffee")
        print("3. Exit")
        
        try:
            option = int(input("Choose an option: "))
            print()

            if option == 1:
                print("Making tea ...")
                tea.prepare_recipe()
            elif option == 2:
                print("Making coffee ...")
                coffee.prepare_recipe()
            elif option == 3:
                print("Exiting the program...")
            else:
                print("Invalid option. Try again.")
        
        except ValueError:
            print("Please enter a valid number only!\n")
        
        print("-" * 30)

File: hw/chavez/u3/hw26TemplateMethod/JavaScript/view/BeverageTest.js
const Tea = require('../model/Tea');
const Coffee = require('../model/Coffee');
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

async function main() {
  let option = 0;
  const tea = new Tea();
  const coffee = new Coffee();

  while (option !== 3) {
    console.log("\n--- Beverage Menu ---");
    console.log("1. Make Tea");
    console.log("2. Make Coffee");
    console.log("3. Exit");

    const input = await new Promise(resolve => {
      rl.question("Choose an option: ", resolve);
    });

    option = parseInt(input.trim());
    console.log();

    switch (option) {
      case 1:
        console.log("Making tea ...");
        await tea.prepareRecipe();
        break;
      case 2:
        console.log("Making coffee ...");
        await coffee.prepareRecipe();
        break;
      case 3:
        console.log("Exiting the program...");
        break;
      default:
        console.log("Invalid option. Try again.");
    }
  }
  rl.close();
}

// Run the application
main().catch(err => console.error("Runtime Error:", err));